rio/src/RIO/Process.hs

```diff
@@ -563,11 +582,16 @@ findExecutable
   -> m (Either ProcessException FilePath) -- ^ Full path to that executable on success
 findExecutable name0 | any FP.isPathSeparator name0 = do
     pc <- view processContextL
-    let names0 = map (name0 ++) (pcExeExtensions pc)
+                 -- On Windows, include the name if it has an existing extension
+    let names0 = (if isWindows && FP.hasExtension name0 then (name0:) else id)
+                 (map (name0 ++) (pcExeExtensions pc))
         testNames [] = return $ Left $ ExecutableNotFoundAt name0
         testNames (name:names) = do
             exists <- liftIO $ D.doesFileExist name
-            if exists
+            existsExec <- if exists
+                              then liftIO $ (isWindows ||) <$> isExecutable name
```

Comment:
I'd rather see `if isWindows then pure True else liftIO $ isExecutable name`. It's clearer in purpose IMO, and doesn't get in the way of shortcircuiting behavior.

Reply:
Will do.